compiler/vm/vmerrors.nim

```diff
@@ -20,10 +20,21 @@ import
 type VmError* = object of CatchableError
   event*: VmEvent
 
-func raiseVmError*(
+func raiseVmErrorImpl(
   event: sink VmEvent;
-  inst:  InstantiationInfo = instLoc()
+  inst:  InstantiationInfo
   ) {.noinline, noreturn.} =
   ## Raises a `VmError`.
   event.instLoc = inst
   raise (ref VmError)(event: event)
+
+# templates below are required as InstantiationInfo isn't captured otherwise
+
+template raiseVmError*(event: sink VmEvent) =
+  ## Raises a `VmError`.
+  {.line.}:
+    raiseVmErrorImpl(event, instLoc(0))
+
+template raiseVmError*(event: sink VmEvent; inst:  InstantiationInfo) =
+  ## Raises a `VmError`.
+  raiseVmErrorImpl(event, inst)
```

Comment:
```suggestion
template raiseVmError*(event: VmEvent) =
  ## Raises a `VmError`, using the source code position of the callsite as the
  ## `inst` value.
  raiseVmErrorImpl(event, instLoc(-2))
```
That should achieve the same thing. `raiseVmErrorImpl` can also be renamed back to `raiseVmError` now.